packages/augur-sdk/src/state/db/BlockAndLogStreamerListener.ts

```diff
@@ -48,93 +76,116 @@ interface LogCallbackMetaData {
 
       callback(blockNumber, parsedLogs);
     };
   }
 
-  listenForEvent(eventNames: string | string[], onLogsAdded: LogCallbackType, onLogsRemoved?: LogCallbackType): void {
-    if(!Array.isArray(eventNames)) eventNames = [eventNames];
+  listenForEvent(
+    eventNames: string | string[],
+    onLogsAdded: LogCallbackType,
+    onLogsRemoved?: LogCallbackType
+  ): void {
+    if (!Array.isArray(eventNames)) eventNames = [eventNames];
 
     // Group by contract address
-    const eventNamesByContractAddress = _.groupBy(eventNames, this.deps.getEventContractAddress);
+    const eventNamesByContractAddress = _.groupBy(
+      eventNames,
+      this.deps.getEventContractAddress
+    );
 
     // For each contract address update the logCallbackMetaData and update the respctive filter for that contract
-    _.forEach(eventNamesByContractAddress, (contractEventNames, contractAddress) => {
-
-      // get all topics for the provided eventNames
-      const topics = contractEventNames.reduce((acc, eventName) => {
-        const topics = this.deps.getEventTopics(eventName);
-        return [
-          ...acc,
-          ...topics,
-        ];
-      }, []);
-
-      // Update the callbacks list with these events and the specified callback
-      this.logCallbackMetaData.push({
-        contractAddress,
-        eventNames: contractEventNames,
-        topics,
-        onLogsAdded: this.filterCallbackByTopic(topics, onLogsAdded)
-      });
-
-      const allContractTopics = this.logCallbackMetaData.reduce((acc, metaData) => {
-        if (metaData.contractAddress == contractAddress) {
-          return [
-            ...acc,
-            ...metaData.topics
-          ]
-        }
-        return acc;
-      }, []);
-
-      // Remove currently active filter if present.
-      if (this.contractFilterUUIDs[contractAddress]) {
-        this.deps.blockAndLogStreamer.removeLogFilter(this.contractFilterUUIDs[contractAddress]);
+    _.forEach(
+      eventNamesByContractAddress,
+      (contractEventNames, contractAddress) => {
+        // get all topics for the provided eventNames
+        const topics = contractEventNames.reduce((acc, eventName) => {
+          const topics = this.deps.getEventTopics(eventName);
+          return [...acc, ...topics];
+        }, []);
+
+        // Update the callbacks list with these events and the specified callback
+        this.logCallbackMetaData.push({
+          contractAddress,
+          eventNames: contractEventNames,
+          topics,
+          onLogsAdded: this.filterCallbackByContractAddressAndTopic(
+            contractAddress,
+            topics,
+            onLogsAdded
+          ),
+        });
       }
-
-      this.contractFilterUUIDs[contractAddress] = this.deps.blockAndLogStreamer.addLogFilter({
-        address: contractAddress,
-        topics: [
-          allContractTopics,
-        ],
-      });
-    });
+    );
   }
 
+  // Note: this assumes the
```

Comment:
unfinished comment?

Reply:
We need more mysterious comments in open source software. Fixed.